homeassistant/components/zwave_js/addon.py

```diff
@@ -55,6 +57,26 @@ async def wrapper(*args, **kwargs):  # type: ignore
     return handle_hassio_api_error
 
 
+@dataclass
+class AddonInfo:
+    """Represent the current add-on info state."""
+
+    options: dict[str, Any]
+    state: AddonState
+    update_available: bool
+    version: str | None
+
+
+class AddonState(Enum):
+    """Represent the current state of the add-on."""
+
+    NOT_INSTALLED = "not_installed"
+    INSTALLING = "installing"
+    UPDATING = "updating"
+    NOT_RUNNING = "not_running"
+    RUNNING = "running"
```

Comment:
Shouldn't this go into the `hassio` integration?

Reply:
Maybe? The add-on info that is returned by the Supervisor API has many properties. These aren't all of them. We'd have to add all of them. Currently three integrations use this Supervisor api: analytics, ozw, zwave_js.

The AddonState currently takes some knowledge from the zwave_js add-on manager. I can remove that, though, I think.

Reply:
> Maybe? The add-on info that is returned by the Supervisor API has many properties. These aren't all of them.

We can always refactor, so it is OK. Just tickling a bit